packages/devtools_app/lib/src/app.dart

```diff
@@ -343,14 +343,9 @@ class DevToolsAppState extends State<DevToolsApp> with AutoDisposeMixin {
             ),
           );
         },
-      if (FeatureFlags.vsCodeSidebarTooling) ..._standaloneScreens,
-    };
-  }
-
-  Map<String, UrlParametersBuilder> get _standaloneScreens {
-    return {
-      for (final type in StandaloneScreenType.values)
-        type.name: (_, __, args, ___) => type.screen,
+      if (FeatureFlags.vsCodeSidebarTooling)
+        for (final type in StandaloneScreenType.values)
+          type.name: (_, __, args, ___) => type.build(_screens),
```

Comment:
@kenzieschmoll I had to inline this to be able to pass screens through - but this doesn't feel very general (other standalone screens might not want screens). Is there something better we could do?

I did wonder if we should move some of the conditions like `requiresDartVm` onto `ScreenMetadata` and then have a full set of those, so we don't actually need the `Screens` to render buttons, only the set of metadata. This could simplify a few places like the screen constructors that all pass name/icon/conditions through to the base - they could just pass their `ScreenMetadata` to the base.

Reply:
This was reverted in favour of the menu just using `ScreenMetaData.values`.